inrupt/pod-browser
# Fix/sharing panel requests

## Description

This PR removes some unnecessary requests done while checking whether a resource is ACP or WAC controlled. 

## Changes

- Refactoring functions to prevent redundant requests
- Removing use of useSWR in access control hook
- Updating usage in relevant components
- Updating tests

## Testing
- Verify resource panel is displayed correctly, including permissions/sharing, both on NSS (keeping in mind the existing access control bug on NSS that causes the permissions panel not to appear in some cases - hard to reproduce consistently) and ESS
- Verify that the number of requests done from the side panel has been reduced

## Commit checklist


## Interested parties

## Notes

## Screenshots/captures
